### DISCUSSION TITLE
Naming issue when generating Ed25519 keys

### DISCUSSION THREAD

**Discussion by gjgd** [panva/jose#88] [Q&A]
Below is how to generate two kinds of keys:

```node
$ node
Welcome to Node.js v12.18.3.
> const jose = require('jose')
> jose.JWK.generate('EC', 'secp256k1').then(console.log)
> ECKey {
  crv: 'secp256k1',
  kid: '83JLNx6uXXTJnLsirAiV9Fc6ZMM-tJeA690EJkSAyo0',
  kty: 'EC',
  x: 'abrHUeLY18TbPRDkRliVUDU1WT3hnAUtLTSQbh2D8pg',
  y: '1xgPpIgij8oQ6IoLctqlnrclBKYmzYRGbj2HSUz9lwM'
}
> jose.JWK.generate('OKP', 'Ed25519').then(console.log)
> OKPKey {
  crv: 'Ed25519',
  kid: 'ozd19EqS-91CgijOLSHXan38SvigBmc3zpNegC22eqU',
  kty: 'OKP',
  x: 'kXmTSBkJhwXNT9ntkRDduniS-LfiNSPU6ZBpevqD7Og'
}
> 
```
The first one makes sense to me: Generate keys from the secp256K1 curve
But the second one uses the term "Ed25519" which I think is not correct:
- [Ed25519](https://ed25519.cr.yp.to/) is the name of the signing algorithm (same as ES256K)
- [Curve25519](https://en.wikipedia.org/wiki/Curve25519) is the name of the curve (same as Secp256K1)

`"crv": "Ed25519"` doesn't make sense to me, I would expect to see
```node
> jose.JWK.generate('OKP', 'Curve25519').then(console.log)
> OKPKey {
  crv: 'Curve25519',
  kid: 'ozd19EqS-91CgijOLSHXan38SvigBmc3zpNegC22eqU',
  kty: 'OKP',
  x: 'kXmTSBkJhwXNT9ntkRDduniS-LfiNSPU6ZBpevqD7Og'
}
> 
```

Thoughts ?

**Comment by panva** [comment]
See, i’m not coming up with the names, i merely implement standards. This is all coming from the respective specifications and its entries in the [IANA registry](https://www.iana.org/assignments/jose/jose.xhtml). Within the JOSE context these are the names given to key’s curves, algs, key types etc. 

All specs are also linked right on top of the readme.
